Graphics/Implicit/Definitions.hs

```diff
@@ -162,6 +165,18 @@ newtype NormedTriangle = NormedTriangle ((ℝ3, ℝ3), (ℝ3, ℝ3), (ℝ3, ℝ3
 -- | A triangle mesh is a bunch of triangles, attempting to be a surface.
 newtype TriangleMesh = TriangleMesh [Triangle]
 
+newtype AnnotatedTriangleMesh a = AnnotatedTriangleMesh { unAnnotatedTriangleMesh :: [(Triangle, a)] }
+
+removeTriangleMeshAnnotations :: AnnotatedTriangleMesh a -> TriangleMesh
+removeTriangleMeshAnnotations (AnnotatedTriangleMesh l) = TriangleMesh $ map fst l
+
+data TriangleProvenance
+   = TriangleProvenance_SquareToTri Bool TriangleProvenance
+   | TriangleProvenance_JoinXAligned TriangleProvenance TriangleProvenance
+   | TriangleProvenance_JoinYAligned TriangleProvenance TriangleProvenance
+   | TriangleProvenance_TesselateLoop Int
```

Comment:
And this int?

Reply:
Yep, just an enum I haven't bothered writing yet.